python/graphstorm/model/loss_func.py

```diff
@@ -294,17 +297,37 @@ def __init__(self, adversarial_temperature):
             "The adversarial_temperature must be a float, but we get " \
             f"{adversarial_temperature} with type {type(adversarial_temperature)}"
         self.adversarial_temperature = adversarial_temperature
+        self._debug_print = True
 
     def _compute_loss(self, score, label):
         """ Compute BCE loss
         """
-        return -(label * th.log(F.sigmoid(score)) +
+        loss = -(label * th.log(F.sigmoid(score)) +
             (1 - label) * th.log(1 - F.sigmoid(score)))
 
+        # For debugging abnormal loss values.
+        if self._debug_print and get_rank() == 0:
+            logging.debug("Cross entropy loss for link prediction \
+                          with input score as {} and loss as {}.".format(
+                          score.detach(), loss.detach()))
+        return loss
+
     def _compute_adversarial_loss(self, score, label):
         loss = self._compute_loss(score, label)
-        loss = th.sum(th.softmax(score * self.adversarial_temperature,
-                                 dim=-1).detach() * loss, dim=-1)
+        loss = th.softmax(score * self.adversarial_temperature,
+                                 dim=-1).detach() * loss
+        # For debugging abnormal loss values.
+        if self._debug_print and get_rank() == 0:
+            logging.debug("Adversarial cross entropy loss for negative samples \
+                          with input score as {} and loss as {}. \
+                          For abnormal loss values, it is suggested to enable \
+                          lp embed normalizer by setting --lp-embed-normalizer l2_norm \
+                          and tune the hyperparameter gamma.".format(
```

Comment:
```suggestion
                          and tune the hyperparameter adversarial_temperature.".format(
```
Tuning gamma does not work.

Reply:
tunning gamma also works.